### ISSUE TITLE
OpenBazaar-IPFS with Tor/I2P

### ISSUE THREAD

**Issue by drwasho** [OpenBazaar/OpenBazaar-Server#362]
The DHT and networking stack of OpenBazaar will be using IPFS. This means that compatibility with Tor/I2P is largely driven by what IPFS supports. I'd like to track work on this front here; so I will be closing previous issues discussing Tor and I2P support in our own networking backend, since we won't be keeping it for much longer.

**Comment by ABISprotocol** [comment]
If I understand correctly, based on this issue and the intended roadmap, OpenBazaar compatibility with Tor/I2P will be driven by what IPFS supports. Right now, IPFS seems to need help with Tor integration. I think this is where people would go to look at that and help out with it, but please correct me if I am wrong or if there are other issue links.
https://www.bountysource.com/issues/26255361-tor-onion-integration

**Comment by drwasho** [comment]
That's correct @ABISprotocol... the good news is that we're not the only ones who want Tor support for IPFS.

**Comment by str4d** [comment]
Just FYI, I2P has launched [a development program](https://geti2p.net/en/blog/post/2016/06/01/I2P-Summer-Dev) aimed at helping developers integrate I2P into their apps. I was originally going to put some work into #240, but since you are moving to IPFS, I'll focus on that instead!

IPFS in Go should be workable with one of the Go libraries for SAM. Python OpenBazaar uses Twisted, so I assume the IPFS implementation used there will also be Twisted-based, which means it should be possible to just drop in txi2p.

**Comment by cpacia** [comment]
@whyrusleeping said he was going to make a tutorial for how to create a custom transport for IPFS so that people could easily follow it to hook up Tor or whatever. Might be worth checking with him to see what the status is.

**Comment by kirkins** [comment]
Interested in hosting a OpenBazaar over i2p. Is it possible currently?

**Comment by ABISprotocol** [comment]
@kirkins Re. I2P, see @str4d's previous comment in this thread, it's been a while since June of 2016 so maybe there's been some progress on that front.  Right now the focus seems to be getting a real release (not just developer release) out for OB with the Tor integration, as I understand it, scheduled for v2.0.  OpenBazaar compatibility with Tor/I2P is driven by what IPFS supports, as I understand it.

**Comment by cpacia** [comment]
Yeah Tor is already working in 2.0. It should probably have a security audit though before people use it.

**Comment by whyrusleeping** [comment]
On I2p, I've talked with @str4d about integrating I2p into libp2p/ipfs. He
had some ideas about making it pretty easy to do. If anyone wants to
attempt that, I'm sure we could advise a best course of action

**Comment by kirkins** [comment]
@whyrusleeping I'm interested in finding out more about what would be needed.
